apps/prairielearn/src/lib/config.ts

```diff
@@ -524,6 +524,13 @@ const ConfigSchema = z.object({
    * to imported questions. Must be true in production for data integrity.
    */
   checkSharingOnSync: z.boolean().default(false),
+  /**
+   * Determines if institution names in course instance access rules should be
+   * validated at sync time. This defaults to false in dev mode where institutions
+   * are not set up, but should be enabled in production to help instructors
+   * catch misconfigured access rules.
+   */
+  checkInstitutionsOnSync: z.boolean().default(false),
```

Comment:
I hate to introduce yet another config option, but I also don't know if `NODE_ENV=production` is a good-enough signal.

Reply:
This will certainly be friendly for not disturbing self-hosters.